x/consensus/keeper/query_queued_messages_for_relaying.go

```diff
@@ -0,0 +1,33 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/palomachain/paloma/x/consensus/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k Keeper) QueuedMessagesForRelaying(goCtx context.Context, req *types.QueryQueuedMessagesForRelayingRequest) (*types.QueryQueuedMessagesForRelayingResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	msgs, err := k.GetMessagesForRelaying(ctx, req.QueueTypeName, req.ValAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	var res []*types.MessageWithSignatures
+	for _, msg := range msgs {
+		if msg.GetRequireSignatures() {
+			res = append(res, k.queuedMessageToMessageWithSignatures(msg))
+		}
+	}
```

Comment:
This is due to the protobuf definition.  As you mentioned in the proposed epic, we can remove the usage of pointers by preventing fields from being null.  However, the [protobuf documentation](https://pkg.go.dev/github.com/gogo/protobuf/gogoproto) states the following:

> Warning about nullable: According to the Protocol Buffer specification, you should be able to tell whether a field is set or unset. With the option nullable=false this feature is lost, since your non-nullable fields will always be set. It can be seen as a layer on top of Protocol Buffers, where before and after marshalling all non-nullable fields are set and they cannot be unset.

It seems to me we should keep these pointers to keep to the standard.  What are your thoughts on that vs the benefits of trying to avoid slices of pointers?

Reply:
I think the warning is perfectly fine on it's own, but doesn't necessarily applied to `repeated` nullable types. And while there is a difference between an empty slice and a `nil` slice logically, I so far have never come across a case where this distinction matters.

But I suggest we move this discussion over to the epic, I just thought I'd point it out in case this was new, independent code. Feel free to resolve this.

Reply:
That's a good point.  It's easy enough to tell if a slice is empty.  I'm happy to make this change.